runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -103,9 +103,15 @@
     public DynamicProtobufSerializer(CorfuRuntime corfuRuntime) {
         this.type = ProtobufSerializer.PROTOBUF_SERIALIZER_CODE;
 
-        // Create and register a protobuf serializer to read the table registry.
-        ISerializer protobufSerializer = createProtobufSerializer();
-        Serializers.registerSerializer(protobufSerializer);
+        // Create or get a protobuf serializer to read the table registry.
+        ISerializer protobufSerializer;
+        try {
+            protobufSerializer = Serializers.getSerializer(ProtobufSerializer.PROTOBUF_SERIALIZER_CODE);
+        } catch (SerializerException se) {
+            // This means the protobuf serializer had not been registered yet.
+            protobufSerializer = new ProtobufSerializer(new ConcurrentHashMap<>());
```

Comment:
Don't we want to register the known types?
classMap.put(getTypeUrl(TableName.getDescriptor()), TableName.class);
        classMap.put(getTypeUrl(TableDescriptors.getDescriptor()),
            TableDescriptors.class);
        classMap.put(getTypeUrl(TableMetadata.getDescriptor()),
            TableMetadata.class);
        classMap.put(getTypeUrl(ProtobufFileName.getDescriptor()),
            ProtobufFileName.class);
        classMap.put(getTypeUrl(ProtobufFileDescriptor.getDescriptor()),
            ProtobufFileDescriptor.class);

Reply:
You're right. I missed that point. Changed to use `createProtobufSerializer` which registers those types.